mesonbuild/build.py

```diff
@@ -110,7 +110,7 @@
 known_exe_kwargs = known_build_target_kwargs | {'implib', 'export_dynamic', 'pie'}
 known_shlib_kwargs = known_build_target_kwargs | {'version', 'soversion', 'vs_module_defs', 'darwin_versions'}
 known_shmod_kwargs = known_build_target_kwargs | {'vs_module_defs'}
-known_stlib_kwargs = known_build_target_kwargs | {'pic', 'prelink'}
+known_stlib_kwargs = known_build_target_kwargs | {'pic', 'pie', 'prelink'}
```

Comment:
I ended up dropping this patch. I'm not sure what the right thing to do is really, this can happen because of magical interactions between b_staticpic and b_pie, but I'm not convinced it's a knob that developers need here.